### PR TITLE
(refactor): kotlin-ify `ShareModuleTest.java`

### PR DESCRIPTION
**Pull request by siddarthkay**
## Summary:

This PR converts the java logic inside of `ShareModuleTest.java` to Kotlin as requested in https://github.com/facebook/react-native/issues/37708 
We also swap `ShareModuleTest.java` for `ShareModuleTest.kt`

## Changelog:

[Internal][Changed]: Convert ShareModuleTest to Kotlin

## Test Plan:

`./gradlew :packages:react-native:ReactAndroid:test` must pass and CI must be green

## Screenshot of passing tests locally 

Screenshot 2023-06-07 at 1 32 22 PM

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/share/ShareModuleTest.kt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+
+@Config(shadows = [ShadowArguments::class])
+@RunWith(RobolectricTestRunner::class)
+class ShareModuleTest {
+
+    private var activity: Activity? = null
+    private var shareModule: ShareModule? = null
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/share/ShareModuleTest.kt:35]:
```suggestion
    private lateinit var activity: Activity
    private lateinit var shareModule: ShareModule
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/share/ShareModuleTest.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        val message = "Message"
+        val dialogTitle = "Dialog Title"
+
+        val content = JavaOnlyMap()
+        content.putString("title", title)
+        content.putString("message", message)
+
+        val promise = SimplePromise()
+
+        shareModule?.share(content, dialogTitle, promise)
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/share/ShareModuleTest.kt:64]: If you use lateinit, all those `?.` can go

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for sending this @siddarthkay 

I've left a minor nit that needs to be addressed before we can merge this. Thanks for doing this work.

**Comment by siddarthkay** [comment]
@cortinico : Thanks for teaching me something new about `Kotlin` :)
I've addressed the changes in https://github.com/facebook/react-native/pull/37746/commits/3160cc99a7772802fbb4e1b1d26f3f058f02d299

**Review by cortinico — APPROVED** [review]
LGTM 👍

**Pull request closed without merging**